### PR TITLE
webPageSummary feedback

### PR DESCRIPTION
**Pull request by TonySchroeder**
close https://github.com/progwise/george-ai/issues/61

- A createSummaryFeedback mutation created in Pothos
- The FeedbackButtons component in Next.js calls it
- And creates a new SummaryFeedback for a WebPageSummary for each click on a thumb
- Display of the thumbs has been simplified

### CODE REVIEW

File: apps/nextjs-web/app/components/info-card/feedback-buttons.tsx [hunk 1]
```diff
@@ -1,45 +1,88 @@
...
+  const [feedbackSelection, setFeedbackSelection] = useState<
+    SummaryFeedbackVoting | undefined
   >()
 
+  const onFeedbackChange = async (votingChoice?: SummaryFeedbackVoting) => {
+    const feedback =
+      feedbackSelection === votingChoice ? undefined : votingChoice
+    setFeedbackSelection(feedback)
+
+    if (feedback) {
```
Comment (moncapitaine) [inline apps/nextjs-web/app/components/info-card/feedback-buttons.tsx:29]:
```suggestion
    if (!feedback) {
       return
    }
```

File: apps/nextjs-web/app/components/info-card/feedback-buttons.tsx [hunk 2]
```diff
@@ -1,45 +1,88 @@
...
-          alt="Thumbs Down outline"
-          className={feedbackChoice === 'down' ? 'opacity-100' : 'opacity-25'}
-          width={24}
-          height={24}
-        />
-      </button>
+      {Object.values(SummaryFeedbackVoting)
+        .reverse()
+        .map((votingChoice) => (
+          <button
```
Comment (moncapitaine) [inline apps/nextjs-web/app/components/info-card/feedback-buttons.tsx:69]: explicit buttons. You never know if you have an image for an value.

File: packages/pothos-graphql/src/WebPageSummary/index.ts [hunk 3]
```diff
@@ -60,8 +60,8 @@ builder.queryField('allSummaries', (t) =>
           `),
           {},
         )
-        const webPageSummarydatas = result.webPageSummaries?.data ?? []
-        return webPageSummarydatas.map((data) => {
+        const webPageSummaryDatas = result.webPageSummaries?.data ?? []
```
Comment (moncapitaine) [inline packages/pothos-graphql/src/WebPageSummary/index.ts:63]: There is no Datas in english language

```suggestion
        const webPageSummariesData = result.webPageSummaries?.data ?? []
```

File: packages/pothos-graphql/src/summaryFeedback/index.ts [hunk 4]
```diff
@@ -0,0 +1,118 @@
+import { builder } from '../builder'
+import { graphql, useFragment } from '../gql'
+import {
+  Enum_Summaryfeedback_Voting,
+  SummaryFeedbackFragment,
+} from '../gql/graphql'
+import { strapiClient } from '../strapi-graphql-client'
+import { formatISO } from 'date-fns'
+
+const SummaryFeedbackVotingReference = builder.enumType(
```
Comment (moncapitaine) [inline packages/pothos-graphql/src/summaryFeedback/index.ts:10]: THis is no reference?

File: packages/pothos-graphql/src/summaryFeedback/index.ts [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+  Enum_Summaryfeedback_Voting,
+  {
+    name: 'SummaryFeedbackVoting',
+  },
+)
+
+const SummaryFeedbackReference = builder.objectRef<SummaryFeedbackFragment>(
```
Comment (moncapitaine) [inline packages/pothos-graphql/src/summaryFeedback/index.ts:17]: Use objectRef again?

File: apps/nextjs-web/app/components/info-card/feedback-buttons.tsx [hunk 6]
```diff
@@ -1,41 +1,98 @@
...
+}
+
+export const FeedbackButtons = ({
+  query,
+  position,
+  webPageSummaryId,
+}: FeedbackButtonsProps) => {
+  const [feedbackSelection, setFeedbackSelection] = useState<
+    SummaryFeedbackVoting | undefined
   >()
```
Comment (PascalHelbig) [inline apps/nextjs-web/app/components/info-card/feedback-buttons.tsx:22]: use `useMutation` and use the result for setting the active state/opacity on the buttons

File: apps/nextjs-web/app/components/info-card/feedback-buttons.tsx [hunk 7]
```diff
@@ -1,41 +1,98 @@
...
+
   return (
     <div className="flex gap-4">
       <button
-        onClick={() =>
-          setFeedbackChoice((previousChoice) =>
-            previousChoice === 'up' ? undefined : 'up',
-          )
-        }
+        key={SummaryFeedbackVoting.Up}
```
Comment (PascalHelbig) [inline apps/nextjs-web/app/components/info-card/feedback-buttons.tsx:68]:
```suggestion
```

File: apps/nextjs-web/app/components/info-card/feedback-buttons.tsx [hunk 8]
```diff
@@ -1,41 +1,98 @@
...
+export const FeedbackButtons = ({
+  query,
+  position,
+  webPageSummaryId,
+}: FeedbackButtonsProps) => {
+  const [feedbackSelection, setFeedbackSelection] = useState<
+    SummaryFeedbackVoting | undefined
   >()
 
+  const onFeedbackChange = async (votingChoice?: SummaryFeedbackVoting) => {
```
Comment (PascalHelbig) [inline apps/nextjs-web/app/components/info-card/feedback-buttons.tsx:24]:
```suggestion
  const handleFeedbackChange = async (votingChoice?: SummaryFeedbackVoting) => {
```

File: apps/nextjs-web/app/components/info-card/feedback-buttons.tsx [hunk 9]
```diff
@@ -1,41 +1,98 @@
...
         />
       </button>
+
       <button
-        onClick={() =>
-          setFeedbackChoice((previousChoice) =>
-            previousChoice === 'down' ? undefined : 'down',
-          )
-        }
+        key={SummaryFeedbackVoting.Down}
```
Comment (PascalHelbig) [inline apps/nextjs-web/app/components/info-card/feedback-buttons.tsx:85]:
```suggestion
```

File: packages/pothos-graphql/src/summaryFeedback/index.ts [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+                query
+                voting
+                web_page_summary {
+                  data {
+                    id
+                  }
+                }
+              }
+            }
+          `),
```
Comment (PascalHelbig) [inline packages/pothos-graphql/src/summaryFeedback/index.ts:106]: This fragment is used in two places (here in the mutation resolver and in the summary type). Lets move it out of this scope to the top of this file.

File: packages/pothos-graphql/src/summaryFeedback/index.ts [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+      resolve: (parent) => parent.attributes?.web_page_summary?.data?.id ?? '',
+    }),
+  }),
+})
+
+builder.mutationField('createSummaryFeedback', (t) =>
+  t.field({
+    type: SummaryFeedbackReference,
+    args: {
+      data: t.arg({ type: CreateSummaryFeedbackInput, required: true }),
```
Comment (PascalHelbig) [inline packages/pothos-graphql/src/summaryFeedback/index.ts:60]: Should be required by default
```suggestion
      data: t.arg({ type: CreateSummaryFeedbackInput }),
```

File: packages/pothos-graphql/src/searchQueryBy/index.ts [hunk 12]
```diff
@@ -51,15 +51,12 @@ builder.queryField('searchResult', (t) =>
         defaultValue: '*',
```
Comment (PascalHelbig) [inline packages/pothos-graphql/src/searchQueryBy/index.ts:51]: Rename file and query field to `searchWebPages`

**Pull request merged**